### PR TITLE
Eddiemmenefee pro 116 add agentnew route

### PR DESCRIPTION
**Pull request by eddie-m-m**
- Add `/agents/new` route to router
- Create `AgentNew` page
- Add preliminary UI elements to `/AgentNew` page (needs a lot of work: some questions in comments on lines 15 and 44)

### CODE REVIEW

**Comment by eddie-m-m** [comment]
Note: `/agents/new` page is not yet wired to 'Add an agent' button in search page.

**Comment by Timmehs** [comment]
> Note: `/agents/new` page is not yet wired to 'Add an agent' button in search page.

The "Add an agent" button will actually need to be `react-router` `<Link />` component with button styling applied. To do this it will need to change to this or similar:
```javascript
            <Link
              to="/agents/new"
              aria-label="Add an agent"
              className="w-75 text-white btn btn-lg btn-primary"
            >
              Add an agent
            </Link>
```

File: src/pages/AgentNew.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      </a>
+      {/* place text horizontal to 'Back' nav link? */}
+      <h2 className="mb-4 fw-normal text-center">Add listing</h2>
+      {/* switch to officer icon */}
+      <div className="mb-4 text-center">
+        <img src={icon} alt="ProPro logo" style={{ width: 50 }} />
+      </div>
+      <div className="p-4 text-start">
+        <h2 className="mb-2">Add an agent</h2>
+        <h5 className="mb-5 fw-normal">
```
Comment (Timmehs) [inline src/pages/AgentNew.tsx:22]: For standard blocks of text, we should use a p tag here
```suggestion
        <p className="mb-5">
```

File: src/pages/AgentNew.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+          <Form.Control placeholder="Agent first name" />
+        </FloatingLabel>
+        <FloatingLabel label="Agent last name" className="mb-4 w-100">
+          <Form.Control placeholder="Agent last name" />
+        </FloatingLabel>
+        <h3 className="mb-5">Office</h3>
+        <div className="mb-5 text-decoration-underline text-center">
+          {/* add url */}
+          <Link to="">Select an office</Link>
+        </div>
```
Comment (Timmehs) [inline src/pages/AgentNew.tsx:39]: Lets keep the vertical spacing consistent with mb-3, and give space around the link with padding.
```suggestion
        <h3 className="mb-3">Office</h3>
        <div className="p-3 mb-3 text-center">
          {/* add url */}
          <Link to="" className="link-dark">
            Select an office
          </Link>
        </div>
```

Comment (Timmehs) [reply src/pages/AgentNew.tsx:39]: Additional note: you may notice the `link-dark` above. Originally I suggested `text-body` but there are special classes for coloring links that support hover/active states: https://getbootstrap.com/docs/5.3/helpers/colored-links/

File: src/pages/AgentNew.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  const navigate = useNavigate()
+
+  return (
+    <div>
+      <a role="button" onClick={() => navigate(-1)}>
+        <i className="bi bi-chevron-left align-middle" />
+        Back
+      </a>
+      {/* place text horizontal to 'Back' nav link? */}
+      <h2 className="mb-4 fw-normal text-center">Add listing</h2>
```
Comment (Timmehs) [inline src/pages/AgentNew.tsx:15]: Can remove

File: src/pages/AgentNew.tsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+      {/* switch to officer icon */}
+      <div className="mb-4 text-center">
+        <img src={icon} alt="ProPro logo" style={{ width: 50 }} />
+      </div>
```
Comment (Timmehs) [inline src/pages/AgentNew.tsx:19]: Here's how I might go about making the icon circle
```suggestion
      <div className="d-flex justify-content-center mb-3">
        <div
          className="d-flex justify-content-center align-items-center bg-white rounded-circle"
          style={{ width: 80, height: 80 }}
        >
          <img src={icon} alt="ProPro logo" width="50%" />
        </div>
      </div>```

**Review by Timmehs — CHANGES_REQUESTED** [review]
It's looking good. HMU if you want to pair or have questions re: on solving the merge conflicts.

File: src/pages/AgentNew.tsx (hunk 4 above)
Comment (eddie-m-m) [reply src/pages/AgentNew.tsx:19]: So helpful! (I had added a to-do comment to figure out the white circle background.) Comparing the size ratios on the app version, it seems the icon should be a bit bigger here, but that's a detail that can be modified later if needed.

File: src/pages/Search.tsx [hunk 5]
```diff
@@ -1,9 +1,9 @@
-import { Form, useLoaderData, useSubmit } from 'react-router-dom'
+import { Form, useLoaderData, useSubmit, Link } from 'react-router-dom'
 import Agent from '../types/Agent'
 import SearchResult from '../components/SearchResult'
 import { useEffect } from 'react'
 import { debounce } from 'lodash'
-import { Button, Card } from 'react-bootstrap'
+import Card from 'react-bootstrap'
```
Comment (Timmehs) [inline src/pages/Search.tsx:6]: You'll still need the destructuring `{}` for this to work.

```suggestion
import { Card } from 'react-bootstrap'
```

Comment (Timmehs) [reply src/pages/Search.tsx:6]: React-bootstrap also recommends importing from the specific file like so, but the lack of auto-complete for this import style makes it tedious.
```
import Card from 'react-bootstrap/Card'
```

https://react-bootstrap.netlify.app/docs/getting-started/introduction#importing-components

Comment (eddie-m-m) [reply src/pages/Search.tsx]: I mixed syntax from those two styles. Good to know which one to go with in any case.

**Comment by eddie-m-m** [comment]
Everything seems to be good now.

**Review by Timmehs — APPROVED** [review]
Looks great!

**Pull request merged**